go-controller/pkg/ovn/ovn_test.go

```diff
@@ -61,7 +69,39 @@ func (o *FakeOVN) init() {
 	o.stopChan = make(chan struct{})
 	o.watcher, err = factory.NewWatchFactory(o.fakeClient)
 	Expect(err).NotTo(HaveOccurred())
-
-	o.controller = NewOvnController(o.fakeClient, o.watcher, o.stopChan, o.asf)
+	o.ovnNBClient = mock.NewMockOVNClient(goovn.DBNB)
+	o.ovnSBClient = mock.NewMockOVNClient(goovn.DBSB)
+	o.controller = NewOvnController(o.fakeClient, o.watcher,
+		o.stopChan, o.asf, o.ovnNBClient,
+		o.ovnSBClient)
 	o.controller.multicastSupport = true
+
+}
+
+func mockAddNBDBError(table, name, field string, err error, ovnNBClient util.OVNInterface) {
+	mockClient, ok := ovnNBClient.(*mock.MockOVNClient)
+	if ok {
+		mockClient.AddToErrorCache(table, name, field, err)
+	}
```

Comment:
`if !ok { panic() }` as that would be a coding error.

Reply:
done.